gems/aws-sdk-core/spec/aws/assume_role_web_identity_credentials_spec.rb

```diff
@@ -143,18 +143,32 @@ module Aws
       c = AssumeRoleWebIdentityCredentials.new(
         role_arn: 'arn',
         web_identity_token_file: token_file_path,
-      ) 
+      )
       expect(c).to be_set
       expect(c.credentials.access_key_id).to eq('akid')
       expect(c.credentials.secret_access_key).to eq('secret')
       expect(c.credentials.session_token).to eq('session')
       expect(c.expiration).to eq(in_one_hour)
     end
 
+    it 'refreshes asynchronously' do
+      allow(credentials).to receive(:expiration).and_return(Time.now)
+      expect(client).to receive(:assume_role_with_web_identity).at_least(2).times
+      expect(File).to receive(:read).with(token_file_path).at_least(2).times
+      expect(Thread).to receive(:new).and_call_original
+
+      c = AssumeRoleWebIdentityCredentials.new(
+        role_arn: 'arn',
+        web_identity_token_file: token_file_path,
+        role_session_name: 'session')
+      c.credentials
+    end
+
     it 'auto refreshes credentials when near expiration' do
       allow(credentials).to receive(:expiration).and_return(Time.now)
-      expect(client).to receive(:assume_role_with_web_identity).exactly(4).times
-      expect(File).to receive(:read).with(token_file_path).exactly(4).times
+      expect(client).to receive(:assume_role_with_web_identity).at_least(4).times
+      expect(File).to receive(:read).with(token_file_path).at_least(4).times
+      expect(Thread).to receive(:new).exactly(3).times.and_call_original
```

Comment:
Potentially might not be needed here 🤷🏽 figured it could be explicit -- but I realize the above test kind of does the same thing with making sure that the message expectations are being called a few times. I'll do more thinking to see if there's a better way to pin the number. I'm not a big fan of using `at_least(N).times` unless it's necessary..